src/anomalib/utils/metrics/perimg/pimo.py

```diff
@@ -0,0 +1,211 @@
+"""Per-Image Overlap (PIMO, pronounced pee-mo) curve.
+
+Two variants of AUCs are implemented:
+    - AUPImO: Area Under the Per-Image Overlap (PIMO) curves.
+              I.e. a metric of per-image average TPR.
+
+for shared fpr = mean( perimg fpr ) == set fpr
+    find the th = fpr^-1( MAX_FPR ) with a binary search on the pixels of the norm images
+    i.e. it's not necessary to compute the perimg fpr curves (tf. binclf curves) in advance
+for other shared fpr alternatives, it's necessary to compute the perimg fpr curves first anyway
+
+further: also choose the th upper bound to be the max score at normal pixels
+"""
+
+
+from __future__ import annotations
+
+from collections import namedtuple
+
+import torch
+from matplotlib.axes import Axes
+from matplotlib.pyplot import Figure
+from numpy import ndarray
+from torch import Tensor
+
+from .binclf_curve import PerImageBinClfCurve
+from .common import (
+    _validate_atleast_one_anomalous_image,
+    _validate_atleast_one_normal_image,
+)
+from .plot import (
+    plot_all_pimo_curves,
+)
+
+# =========================================== METRICS ===========================================
+
+PImOResult = namedtuple(
```

Comment:
do you have any specific motivation why use named tuple instead of data class. I was just curious :)  Just preference or any other motivation?

Reply:
Yes and more or less. 

It's possible to just unpack the things, like

```python
_, __, shared_fpr, tprs, image_classes = self.compute()
```.

Also it's just a tuple, I think it makes user code less dependent on in-`anomalib` classes. I tend to use it for this type of case where I actually just want to return many things but there are so many that it's easier to document it as a namedtuple. I noticed `scipy` does something similar (?).

Otherwise I'm a big fan of dataclass : )